pkg/blockbuilder/scheduler/strategy.go

```diff
@@ -63,7 +63,12 @@ func (p *RecordCountPlanner) Plan(ctx context.Context, maxJobsPerPartition int,
 		// 1. kadm.GroupMemberLag contains valid Commit.At even when consumer group never committed any offset.
 		//    no additional validation is needed here
 		// 2. committed offset could be behind start offset if we are falling behind retention period.
+
+		// startOffset is the previously-committed offset, so we need to start processing the first
+		// _uncommitted_ offset
 		startOffset := max(partitionOffset.Commit.At+1, partitionOffset.Start.Offset)
+		// Likewise, endOffset is initially the next available offset: this is why we treat jobs as end-exclusive:
```

Comment:
shouldn't we include the `endOffset` since it is a valid offset for a record in the partition? we'd be skipping the last record otherwise

Reply:
WDYM? end offset in this case is _exclusive_. This is the next available record in partitions which we're caught up on. Trying to poll for these when the partition is no longer being written to causes us to hang forever waiting for data which will not appear. This is the reason we were seeing infinite polling on defunct partitions.